sentry-ruby/spec/sentry/transport/http_transport_spec.rb

```diff
@@ -129,6 +129,21 @@
       subject.send_data(data)
     end
 
+    it "accepts a proxy from ENV[HTTP_PROXY]" do
+      ENV["http_proxy"] = "https://stan:foobar@example.com:8080"
+
+      stub_request(fake_response) do |_, http_obj|
+        expect(http_obj.proxy_address).to eq("example.com")
+        expect(http_obj.proxy_user).to eq("stan")
+        expect(http_obj.proxy_pass).to eq("foobar")
+        expect(http_obj.proxy_port).to eq(8080)
+      end
+
+      subject.send_data(data)
+
+      ENV["http_proxy"] = nil 
```

Comment:
This need to be put inside an ensure block so when it failed the env wouldn't leak to other tests.

```suggestion
      begin
        ENV["http_proxy"] = "https://stan:foobar@example.com:8080"
  
        stub_request(fake_response) do |_, http_obj|
          expect(http_obj.proxy_address).to eq("example.com")
          expect(http_obj.proxy_user).to eq("stan")
          expect(http_obj.proxy_pass).to eq("foobar")
          expect(http_obj.proxy_port).to eq(8080)
        end
  
        subject.send_data(data)
      ensure
        ENV["http_proxy"] = nil
      end
```